x-pack/plugins/cloud_security_posture/public/common/component/multi_select_filter.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { css } from '@emotion/react';
+import type { EuiSelectableOption } from '@elastic/eui';
+import {
+  EuiPopoverTitle,
+  EuiCallOut,
+  EuiHorizontalRule,
+  EuiPopover,
+  EuiSelectable,
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiText,
+} from '@elastic/eui';
+import { isEqual } from 'lodash/fp';
+import { i18n } from '@kbn/i18n';
+
+type FilterOption<T extends string, K extends string = string> = EuiSelectableOption<{
+  key: K;
+  label: T;
+}>;
+
+export type { FilterOption as MultiSelectFilterOption };
+
+export const mapToMultiSelectOption = <T extends string>(options: T[]) => {
+  return options.map((option) => {
+    return {
+      key: option,
+      label: option,
+    };
+  });
+};
+
+const fromRawOptionsToEuiSelectableOptions = <T extends string, K extends string>(
+  options: Array<FilterOption<T, K>>,
+  selectedOptionKeys: string[]
+): Array<FilterOption<T, K>> => {
+  return options.map(({ key, label }) => {
+    const selectableOption: FilterOption<T, K> = { label, key };
+    if (selectedOptionKeys.includes(key)) {
+      selectableOption.checked = 'on';
+    }
+    selectableOption['data-test-subj'] = `options-filter-popover-item-${key.split(' ').join('-')}`;
+    return selectableOption;
+  });
+};
+
+const fromEuiSelectableOptionToRawOption = <T extends string, K extends string>(
+  options: Array<FilterOption<T, K>>
+): string[] => {
+  return options.map((option) => option.key);
+};
+
+const getEuiSelectableCheckedOptions = <T extends string, K extends string>(
+  options: Array<FilterOption<T, K>>
+) => options.filter((option) => option.checked === 'on') as Array<FilterOption<T, K>>;
+
+interface UseFilterParams<T extends string, K extends string = string> {
+  buttonIconType?: string;
+  buttonLabel?: string;
+  hideActiveOptionsNumber?: boolean;
+  id: string;
+  limit?: number;
+  limitReachedMessage?: string;
+  onChange: (params: { filterId: string; selectedOptionKeys: string[] }) => void;
+  options: Array<FilterOption<T, K>>;
+  renderOption?: (option: FilterOption<T, K>) => React.ReactNode;
+  selectedOptionKeys?: string[];
+  transparentBackground?: boolean;
+}
+export const MultiSelectFilter = <T extends string, K extends string = string>({
```

Comment:
please add a comment with a link to the original code so we can reference to it in the future if we need to